main.tf

```diff
@@ -1,28 +1,35 @@
 data "aws_caller_identity" "current" {}
 
-data "aws_sns_topic" "bucket-arn" {
-  name = "bucket-arn"
+resource "aws_s3_bucket" "bucket" {
+  bucket = aws_s3_bucket.bucket.id
 }
 
+resource "aws_s3_bucket_notification" "bucket_notification" {
+  bucket = aws_s3_bucket.bucket.id
+
+  topic {
+    topic_arn = var.sns_arn
+    #events        = ["s3:ObjectCreated:*"]
```

Comment:
This is a good start, but the `events` attribute is mandatory, so you will need that being passed as a variable as well.